mungegithub/mungers/submit-queue.go

```diff
@@ -159,8 +160,14 @@ type submitQueueInterruptedObject struct {
 type submitQueueMetadata struct {
 	ProjectName string
 
-	ChartUrl    string
-	HistoryUrl  string
+	ChartUrl   string
+	HistoryUrl string
+	// chartUrl and historyUrl are option storage locations. They are distinct from ChartUrl and
+	// HistoryUrl since the the public variables are used asynchronously by a fileserver and updates
+	// to the options values should not cause a race condition.
```

Comment:
Strings are immutable, but from what I can tell they are defined in a struct greater than the word size of the machine. (see: https://groups.google.com/forum/#!topic/golang-nuts/ZRKSJ3GPkLw and http://www.tapirgames.com/blog/golang-string) Copying one of these structs wouldn't necessarily be an atomic memory operation so a read and a write at the same time wouldn't be safe because the struct could be partially updated?